blast/ncbi/doc/man/makeset.1
.TH MAKESET 1 2001-10-05 NCBI "NCBI Tools User's Manual"
.SH NAME
makeset \- build a Bioseq-set from Seq-entries
.SH SYNOPSIS
.B makeset
[\|\fB\-\fP\|]
[\|\fB\-b\ F\fP\|]
[\|\fB\-c\fP\ \fIN\fP\|]
[\|\fB\-d\fP\|]
[\|\fB\-i\fP\ \fIfilename\fP\|]
[\|\fB\-o\fP\ \fIfilename\fP\|]
[\|\fB\-r\fP\ \fIstr\fP\|]
.SH DESCRIPTION
\fBmakeset\fP builds an ASN.1 Bioseq-set from a bunch of files
containing Seq-entry objects.
.SH OPTIONS
A summary of options is included below.
.TP
\fB\-\fP
Print usage message
.TP
\fB\-b\ F\fP
Output data as text, not binary
.TP
\fB\-c\fP\ \fIN\fP
Bioseq-set.class (default = \fB255\fP ["other"])
.TP
\fB\-d\fP
Input data is binary
.TP
\fB\-i\fP\ \fIfilename\fP
Read list of files from \fIfilename\fP (default = \fBstdin\fP)
.TP
\fB\-o\fP\ \fIfilename\fP
Output Bioseq-set (default = \fBstdout\fP)
.TP
\fB\-r\fP\ \fIstr\fP
Bioseq-set.release
.SH AUTHOR
The National Center for Biotechnology Information.


QuickTimeStreamingServer/ReleaseNotes.txt

                                 RELEASE NOTES
                             Darwin Streaming Server 

These release notes are for the current development version of Darwin
Streaming Server.

------------------------------------------------------------------------
Date: 8/7/2001
Version: 3.0.1+development [v349]
cvs tag: qtssServer-349

*   Updated build platforms: Solaris 8, Redhat Linux 7.1, FreeBSD 4.3, 
    Mac OS X 10.1
*   Retransmit/overbuffering (Skip Protection) algorithms tuned
    substantially
*   "reliable_udp" attribute defaults to "enabled" 
*   Various WebAdmin fixes for Windows platform

------------------------------------------------------------------------
Date: 8/9/2001
Version: 3.0.1+development [v350]
cvs tag: qtssServer-350

*   Minor bug fixes.

------------------------------------------------------------------------
Date: 8/14/2001
Version: 3.0.1+development [v351]
cvs tag: qtssServer-351

*   QTSSFileModule - enabled caching based on preferences (shared and
    private).
*   Add buffer delay to on demand movies.
*   QTFile - code cleanup.
*   Changed some paths for the installer.
*   Added support for MP3Broadcaster on Windows.

------------------------------------------------------------------------
Date: 8/16/2001
Version: 3.0.1+development [v352]
cvs tag: qtssServer-352

*   Added MP3StreamingModule to all builds.
*   Fixed gdb script to execute QTSS from build dir on MacOSX.

------------------------------------------------------------------------
Date: 8/19/2001
Version: 3.0.1+development [v353]
cvs tag: qtssServer-353

*   Added:
        add_seconds_to_client_buffer_delay -- Increases client buffer
                                delay before playing movie.
        record_movie_file_sdp -- Records a .sdp file for a .mov file.
                                Appears as moviename.mov.sdp
        enable_movie_file_sdp -- Plays a .mov.sdp file for a .mov movie.
*   Fixed server to play .mov.sdp files.
*   QTSSFileModule enables both private and shared movie cache buffers.

------------------------------------------------------------------------
Date: 8/23/2001
Version: 3.0.1+development [v354]
cvs tag: qtssServer-354

*   Added prefs for MP3StreamingModule.
*   Default Reliable UDP dir is / on UNIX and \ on NT
*   Use Enable Reliable UDP pref to disable Reliable UDP
*   Included new Web Admin
*   Changed relay preferences to XML format.
        Old relay config files will not work.

------------------------------------------------------------------------
Date: 08/28/2001
Version: 3.0.1+development [v355]
cvs tag: qtssServer-355

*	Added preference:
	 RTSP_error_message to enable debugging error strings returned in 
	 the content body of RTSP error responses.
	
*	Changed RTSP code to:
	1) require a Cseq field for OPTIONS requests
	2) allow missing Range headers in a PLAY request (defaults to entire movie)
	3) add content length header to Error responses containing error messages
	4) add SDP o= and t= fields to on-demand movie Describe responses

*      Changed Relay to relay incoming ANNOUNCEd broadcasts
       1) All ANNOUNCEd broadcasts, or
       2) All broadcasts from an IP address
       3) A broadcast URL from an IP address

       (a) Can still configure the relay to relay UDP streams
       (b) Can pull an SDP or movie from source and relay it

*      Changed Relay to ANNOUNCE the SDPs to its destinations
       1) for ANNOUNCED sources
       2) for sources that are pulled from the source server

*      Changed Relay config file to be in XML format
       1) relayconfig.xml-Sample will be installed instead of old streamingrelay.conf

*	Added General Settings and Change Password to UI
	1) Require existing password to change to new
	2) Require the user to enter the same password twice
	3) Change the auth cookie so the user won't have to re-authenticate

-------------------------------------------------------------------------------------------------
Date: 8/31/2001
Version: 3.0.1+development [v356]
cvs tag: qtssServer-356

*	 Admin Module change
	 1) Added an admin module pref called "AdministratorGroup" with
	 a default value of "admin"
	 2) Changed Admin Module to allow all users that belong to the
	 AdministratorGroup, instead of allowing only one user called
	 "streamingadmin".
	 3) A default qtgroups gets installed from now. The default qtusers
	 file has user 'streamingadmin' with password 'default'. The default
	 qtgroups file has the group 'admin' with the user 'streamingadmin'.
	 4) The login name for the web admin can be changed by adding the user
	 to the qtusers file and adding the user to the 'admin' group in the
	 qtgroups file.

*	Added "RTSP_debug_printfs" server pref to show the incoming and outgoing RTSP.
	Fixed Session header response to remove ";timeout=" field if the value is not defined.
	Fixed .mov.sdp Describe response to include the Last-Modified and Cache-Control headers.
	Fixed RTSP Session header parser to also stop on a ';'
	Changes shared file buffer size to default to maximum size if the bit-rate is 0 and no fixed 
	buffer size is defined.

-------------------------------------------------------------------------------------------------
Date: 9/04/2001
Version: 3.0.1+development [v357]
cvs tag: qtssServer-357

*	more implementation code for Web admin added.
*	New XML parser for reading prefs.
*	full implementation of MP3 streaming module is included.

-------------------------------------------------------------------------------------------------
Date: 9/06/2001
Version: 3.0.1+development [v359]
cvs tag: qtssServer-359

*	Relay that pulls source from the same server can be setup now
	[rtsp_source SOURCE with own IP address as the source_addr]
   
-------------------------------------------------------------------------------------------------
Date: 9/10/2001
Version: 3.0.1+development [v361]
cvs tag: qtssServer-361

*	UI: play and stop buttons in playlist screen now start/stop playlists immediately

blast/ncbi/demo/makeset.c
/*****************************************************************************
*
*   makeset.c
*      build a Bioseq-set from a bunch of Seq-entry files
*      expects a file of file names in stdin
*      prints the Bioseq-set out stdout in binary
*
*****************************************************************************/
#include <all.h>

#define NUMARGS 6
Args myargs[NUMARGS] = {
	{ "Input File Names", "stdin", NULL, NULL, FALSE, 'i', ARG_FILE_IN, 0.0,0,NULL},
	{ "Input data is binary", "F", NULL, NULL, TRUE , 'd', ARG_BOOLEAN, 0.0,0,NULL},
	{ "Output data as binary", "T", NULL, NULL, TRUE , 'b', ARG_BOOLEAN, 0.0,0,NULL},
	{ "Output Bioseq-set", "stdout", "Bioseq-set", NULL, FALSE, 'o', ARG_DATA_OUT, 0.0,0,NULL},
	{ "Bioseq-set.class", "255", NULL, NULL, FALSE , 'c', ARG_INT, 0.0,0,NULL},
	{ "Bioseq-set.release", NULL, NULL, NULL, TRUE , 'r', ARG_STRING, 0.0,0,NULL}};



Int2 Main(void)
{
	AsnIoPtr aipin, aip;
	AsnTypePtr atp;
	DataVal value;
	Int4 seekptr, tempseek, uid;
	static CharPtr outtypes[2] = { "w", "wb" };
	static CharPtr intypes[2] = { "r", "rb" };
	Int2 outtype, intype;
	FILE *fp;
	Char fname[80];
	CharPtr tmp;

    if (! AsnLoad())
        Message(MSG_FATAL, "Unable to load all parse tree.");

	if (! GetArgs("MakeSet 1.0", NUMARGS, myargs))
		return 1;

	if (myargs[1].intvalue)        /* binary input is TRUE */
		intype = 1;
	else
		intype = 0;

	if (myargs[2].intvalue)        /* binary output is TRUE */
		outtype = 1;
	else
		outtype = 0;

	if ((aip = AsnIoOpen(myargs[3].strvalue, outtypes[outtype])) == NULL)
	{
		Message(MSG_ERROR, "Couldn't open %s for output", myargs[3].strvalue);
		return 1;
	}

	if ((fp = FileOpen(myargs[0].strvalue, "r")) == NULL)
	{
		Message(MSG_ERROR, "Couldn't open %s for input", myargs[0].strvalue);
		return 1;
	}

	AsnOpenStruct(aip, BIOSEQ_SET, NULL);
	value.intvalue = myargs[4].intvalue;
	AsnWrite(aip, BIOSEQ_SET_class, &value);
	if (myargs[5].strvalue != NULL)
	{
		value.ptrvalue = myargs[5].strvalue;
		AsnWrite(aip, BIOSEQ_SET_release, &value);
	}
	AsnOpenStruct(aip, BIOSEQ_SET_seq_set, NULL);

	while (FileGets(fname, (size_t)78, fp) != NULL)
	{
		tmp = fname;
		if (*tmp <= ' ') break;
		while (*tmp >= ' ') tmp++;
		*tmp = '\0';
		aipin = AsnIoOpen(fname, intypes[intype]);
		atp = SEQ_ENTRY;
		atp = AsnReadId(aipin, amp, atp);
		AsnReadVal(aipin, atp, &value);
		AsnWrite(aip, BIOSEQ_SET_seq_set_E, &value);
		while (AsnGetLevel(aipin) > 0)
		{
			atp = AsnReadId(aipin, amp, atp);
			AsnReadVal(aipin, atp, &value);
			AsnWrite(aip, atp, &value);
			AsnKillValue(atp, &value);
		}
		aipin = AsnIoClose(aipin);
	}
	AsnCloseStruct(aip, BIOSEQ_SET_seq_set, NULL);
	AsnCloseStruct(aip, BIOSEQ_SET, NULL);
	aip = AsnIoClose(aip);
	FileClose(fp);
	return 0;
}


